platform/darwin/src/MGLGeometry_Private.h

```diff
@@ -123,3 +123,48 @@ NS_INLINE MGLRadianCoordinate2D MGLRadianCoordinateAtDistanceFacingDirection(MGL
                                                          cos(distance) - sin(coordinate.latitude) * sin(otherLatitude));
     return MGLRadianCoordinate2DMake(otherLatitude, otherLongitude);
 }
+
+/** Returns the coordinate at a given fraction between two coordinates.
+ 
+ @param origin Origin `CLLocationCoordinate2D`.
+ @param destination Destination `CLLocationCoordinate2D`.
+ @param fraction Fraction between coordinates (0 = origin, 0.5 = middle, 1 = destination).
+ 
+ @return A coordinate at given fraction. */
+// Ported from http://www.movable-type.co.uk/scripts/latlong.html
+NS_INLINE CLLocationCoordinate2D MGLCLCoordinateAtFraction(CLLocationCoordinate2D origin, CLLocationCoordinate2D destination, double fraction) {
```

Comment:
This is a lot of code to inline. Move the body of the function to MGLGeometry.mm.